scipy/sparse/construct.py

```diff
@@ -770,10 +768,32 @@ def random(m, n, density=0.01, format='coo', dtype=None,
     elif isinstance(random_state, (int, np.integer)):
         random_state = np.random.RandomState(random_state)
     if data_rvs is None:
-        data_rvs = random_state.rand
+        if np.issubdtype(dtype, np.integer):
+            try:
+                np.lib.NumpyVersion
+            except AttributeError:
+                is_old_numpy_version = True
+            else:
+                numpy_version = np.lib.NumpyVersion(np.__version__)
+                is_old_numpy_version = numpy_version < '1.11.0'
+            if not is_old_numpy_version:
```

Comment:
Perhaps this random_integers function could be added to https://github.com/scipy/scipy/blob/master/scipy/_lib/_numpy_compat.py ?

Reply:
The `RandomState` part of NumPy is written in Cython.

Reply:
Sorry, what I mean was: add a `randint` function to `_numpy_compat` that accepts a `dtype` kwarg, so we don't have to do the version check every time `scipy.sparse.random()` is called. It might look like:

```python
if ...numpy version check is good...:
  randint = np.random.randint
else:
  def randint(*args, **kwargs):
    kwargs.pop('dtype', None)  # Ignore dtype.
    return np.random.randint(*args, **kwargs)
```

Reply:
Okay, I have added a patched version of `np.random` to `scipy._lib._numpy_compat` and us this one now.